tests/integration/s2n_resume_test.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+import os
+import sys
+import time
+import ssl
+import socket
+import subprocess
+
+# All supported ciphers
+S2N_CIPHERS= [
+    ("RC4-MD5", ssl.PROTOCOL_SSLv3),
+    ("RC4-SHA", ssl.PROTOCOL_SSLv3),
+    ("DES-CBC3-SHA", ssl.PROTOCOL_SSLv3),
+    ("EDH-RSA-DES-CBC3-SHA", ssl.PROTOCOL_SSLv3),
+    ("AES128-SHA", ssl.PROTOCOL_TLSv1),
+    ("DHE-RSA-AES128-SHA", ssl.PROTOCOL_TLSv1),
+    ("AES256-SHA", ssl.PROTOCOL_TLSv1),
+    ("DHE-RSA-AES256-SHA", ssl.PROTOCOL_TLSv1),
+    ("AES128-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("AES256-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("DHE-RSA-AES128-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("DHE-RSA-AES256-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("AES128-GCM-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("AES256-GCM-SHA384", ssl.PROTOCOL_TLSv1_2),
+    ("DHE-RSA-AES128-GCM-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("ECDHE-RSA-DES-CBC3-SHA", ssl.PROTOCOL_TLSv1),
+    ("ECDHE-RSA-AES128-SHA", ssl.PROTOCOL_TLSv1),
+    ("ECDHE-RSA-AES256-SHA", ssl.PROTOCOL_TLSv1),
+    ("ECDHE-RSA-AES128-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("ECDHE-RSA-AES256-SHA384", ssl.PROTOCOL_TLSv1_2),
+    ("ECDHE-RSA-AES128-GCM-SHA256", ssl.PROTOCOL_TLSv1_2),
+    ("ECDHE-RSA-AES256-GCM-SHA384", ssl.PROTOCOL_TLSv1_2),
+]
+
+PROTO_VERS_TO_STR = {
+    ssl.PROTOCOL_SSLv3 : "SSLv3",
+    ssl.PROTOCOL_TLSv1 : "TLSv1.0",
+    ssl.PROTOCOL_TLSv1_1 : "TLSv1.1",
+    ssl.PROTOCOL_TLSv1_2 : "TLSv1.2",
+}
+
+PROTO_VERS_TO_ARG = {
+    ssl.PROTOCOL_SSLv3 : "-ssl3",
+    ssl.PROTOCOL_TLSv1 : "-tls1",
+    ssl.PROTOCOL_TLSv1_1 : "-tls1_1",
+    ssl.PROTOCOL_TLSv1_2 : "-tls1_2",
+}
+
+def try_resume(endpoint, port, cipher, ssl_version):
+    # Fire up s2nd
+    s2nd = subprocess.Popen(["../../bin/s2nd", "-c", "test_all", str(endpoint), str(port)], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    # Make sure it's running
+    s2nd.stdout.readline()
+
+    # Fire up s_client
+    s_client = subprocess.Popen(["../../libcrypto-root/bin/openssl", "s_client", PROTO_VERS_TO_ARG[ssl_version], "-cipher", cipher,
+                                 "-quiet", "-reconnect", "-connect", str(endpoint) + ":" + str(port)], stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                 stderr=subprocess.DEVNULL)
+
+    # Wait until the 6th connection
+    seperators = 0
+    for line in s2nd.stdout:
+        line = line.decode("utf-8").strip()
+        if line.startswith("Cipher negotiated"):
```

Comment:
doh this is the s2nd output, not s_client. We could add an accessor for resumption status that s2nd can query. Something like `uint8_t s2n_conn_is_session_resumed(struct s2n_connection *conn)`?